docs/api/matchers.md

```diff
@@ -42,6 +42,20 @@ Match elements with the specified text.
 element(by.text('Tap Me'));
 ```
 
+You can also use regex with support of the following JS flags:
+
+| Flag | Name          | Modification                                                                                                                                       |
+|------|---------------|----------------------------------------------------------------------------------------------------------------------------------------------------|
+| `i`  | Ignore Casing | Makes the expression search case-insensitively.                                                                                                    |
+| `s`  | Dot All       | Makes the wild character . match newlines as well.                                                                                                 |
+| `m`  | Multiline     | Makes the boundary characters ^ and $ match the beginning and ending of every single line instead of the beginning and ending of the whole string. |
+
+**Note:** JS flags like `g` (global) and `y` (sticky) that are not supported, as well as `u` (unicode) which is always implied, are ignored when parsing input.
+
+```js
+element(by.text(/Tap [A-Za-z]+/i));
```

Comment:
I love it. Minor suggestion for improvement:

```suggestion
You can also utilize regex (Regular Expressions) alongside certain flags. Here's a table with the supported flags:

| Flag | Name          | Modification                                                                                                                                       |
|------|---------------|----------------------------------------------------------------------------------------------------------------------------------------------------|
| `i`  | Ignore Casing | Makes the expression search case-insensitively.                                                                                                    |
| `s`  | Dot All       | Makes the wild character `.` match newlines as well.                                                                                                 |
| `m`  | Multiline     | Makes the boundary characters (`^` and `$`) match the beginning and ending of every single line instead of the beginning and ending of the whole string. |

**Note:** JS flags like `g` (global) and `y` (sticky) that are not supported, as well as `u` (unicode) which is always implied, are ignored when parsing input.

The following sample code snippet matches text starting with "Tap" followed by any number of alphabetic characters, case-insensitively:

```js
element(by.text(/Tap [A-Za-z]+/i));
```